### ISSUE TITLE
java.lang.ClassCastException: com.intel.analytics.zoo.models.image.objectdetection.ssd.SSDVGG cannot be cast to com.intel.analytics.zoo.models.image.objectdetection.ObjectDetector

### ISSUE THREAD

**Issue by beta21s** [intel-analytics/analytics-zoo#3299]
I have performed the training as instructed above [https://github.com/intel-analytics/analytics-zoo/tree/master/zoo/src/main/scala/com/intel/analytics/zoo/examples/objectdetection/finetune/ssd](https://github.com/intel-analytics/analytics-zoo/tree/master/zoo/src/main/scala/com/intel/analytics/zoo/examples/objectdetection/finetune/ssd) 
And exported the network model with the ```final.model``` file. But when running the demo to detect object from re-train model, an error occurs 
```
py4j.protocol.Py4JJavaError: An error occurred while calling o45.loadObjectDetector.
: java.lang.ClassCastException: com.intel.analytics.zoo.models.image.objectdetection.ssd.SSDVGG cannot be cast to com.intel.analytics.zoo.models.image.objectdetection.ObjectDetector
```

Code re-train model
```
spark-submit \
    --master local[4] \
    --driver-memory 20g \
    --executor-memory 20g \
    --conf "spark.serializer=org.apache.spark.serializer.JavaSerializer" \
    --class com.intel.analytics.zoo.examples.objectdetection.finetune.ssd.Train \
    analytics-zoo-bigdl_0.12.1-spark_2.4.3-0.9.0-jar-with-dependencies.jar --resolution 300 --model analytics-zoo_ssd-mobilenet-300x300_PASCAL_0.1.0.model \
    --class /home/at/Desktop/xe/classname.txt -f /home/at/Desktop/xe/seq/train -v /home/at/Desktop/xe/seq/test -e 1 -b 8 --modelDir /home/at/Desktop/xe/model -l 0.0001 --learningRateDecay 0.0005
```

This is my code
```
import argparse
import cv2

from zoo.common.nncontext import init_nncontext
from zoo.models.image.objectdetection import *

sc = init_nncontext("Object Detection Example")
def predict(model_path, img_path, output_path, partition_num):
    model = ObjectDetector.load_model(model_path)
    model = model.predict(model)
    image_set = ImageSet.read(img_path, sc, image_codec=1, min_partitions=partition_num)
    output = model.predict_image_set(image_set)

    config = model.get_config()
    visualizer = Visualizer(config.label_map(), encoding="jpg")
    visualized = visualizer(output).get_image(to_chw=False).collect()
    for img_id in range(len(visualized)):
        cv2.imwrite(output_path + '/' + str(img_id) + '.jpg', visualized[img_id])

if __name__ == "__main__":
    model_path='/home/at/Desktop/final.model'
    img_path='/home/at/Desktop/img-3.png'
    output_path = '/home/at/Desktop/'
    partition_num = 1
    predict(model_path, img_path, output_path, partition_num)
print("finished...")
sc.stop()
```

I think it's not the correct class. But I don't know how to do it.
Can you guide me?
Sincerely thank

**Comment by helenlly** [comment]
@jenniew  pls have a look

**Comment by qiuxin2012** [comment]
@beta21s , it seems `model = model.predict(model)`(line 10 in your code) is a wrong usage, please delete it.

And could you show me the full error stack under `py4j.protocol.Py4JJavaError: An error occurred while calling o45.loadObjectDetector.
: java.lang.ClassCastException: com.intel.analytics.zoo.models.image.objectdetection.ssd.SSDVGG cannot be cast to com.intel.analytics.zoo.models.image.objectdetection.ObjectDetector`?

**Comment by jenniew** [comment]
Hi @beta21s , I have fixed this issue. Please get the latest build of Analytics Zoo to try. 
When you fine tune ssdvgg model, you'd better pass pretrained analytics zoo sddvgg model not the mobilenet model to load corresponding weights. After you finetune the sddvgg model, you can call `model = ObjectDetector.load_model(model_path)` to load finetuned model, and create your own `ImageConfigure` object then call `model.predict_image_set(image,  image_configure)`. You cannot use `model.predict_image_set(image_set)` directly as we only support to create default image configuration for models published by analytics zoo not your fine-tuned model. You need to pass your own image configuration during prediction. For how to create `ImageConfigure`, you can refer https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/test/zoo/pipeline/api/keras/test_simple_integration.py#L180

**Comment by beta21s** [comment]
> Hi @beta21s , I have fixed this issue. Please get the latest build of Analytics Zoo to try.
> When you fine tune ssdvgg model, you'd better pass pretrained analytics zoo sddvgg model not the mobilenet model to load corresponding weights. After you finetune the sddvgg model, you can call `model = ObjectDetector.load_model(model_path)` to load finetuned model, and create your own `ImageConfigure` object then call `model.predict_image_set(image, image_configure)`. You cannot use `model.predict_image_set(image_set)` directly as we only support to create default image configuration for models published by analytics zoo not your fine-tuned model. You need to pass your own image configuration during prediction. For how to create `ImageConfigure`, you can refer https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/test/zoo/pipeline/api/keras/test_simple_integration.py#L180

**Comment by beta21s** [comment]
> > Hi @beta21s , I have fixed this issue. Please get the latest build of Analytics Zoo to try.
> > When you fine tune ssdvgg model, you'd better pass pretrained analytics zoo sddvgg model not the mobilenet model to load corresponding weights. After you finetune the sddvgg model, you can call `model = ObjectDetector.load_model(model_path)` to load finetuned model, and create your own `ImageConfigure` object then call `model.predict_image_set(image, image_configure)`. You cannot use `model.predict_image_set(image_set)` directly as we only support to create default image configuration for models published by analytics zoo not your fine-tuned model. You need to pass your own image configuration during prediction. For how to create `ImageConfigure`, you can refer https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/test/zoo/pipeline/api/keras/test_simple_integration.py#L180

@jenniew  Thank you for your help, I will reinstall

**Comment by beta21s** [comment]
@jenniew I clone the project again and ```bash make-dist.sh``` generates this error. Hope for a help

```
[WARNING]  Expected all dependencies to require Scala version: 2.11.8
[WARNING]  org.scalatest:scalatest_2.11:3.0.7 requires scala version: 2.11.12
[WARNING] Multiple versions of scala libraries detected!
[INFO] Using incremental compilation
[INFO] 'compiler-interface' not yet compiled for Scala 2.11.8. Compiling...
error: scala.reflect.internal.MissingRequirementError: object java.lang.Object in compiler mirror not found.
	at scala.reflect.internal.MissingRequirementError$.signal(MissingRequirementError.scala:17)
	at scala.reflect.internal.MissingRequirementError$.notFound(MissingRequirementError.scala:18)
	at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:53)
	at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:45)
	at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:45)
	at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:66)
	at scala.reflect.internal.Mirrors$RootsBase.getClassByName(Mirrors.scala:102)
	at scala.reflect.internal.Mirrors$RootsBase.getRequiredClass(Mirrors.scala:105)
	at scala.reflect.internal.Definitions$DefinitionsClass.ObjectClass$lzycompute(Definitions.scala:257)
	at scala.reflect.internal.Definitions$DefinitionsClass.ObjectClass(Definitions.scala:257)
	at scala.reflect.internal.Definitions$DefinitionsClass.init(Definitions.scala:1394)
	at scala.tools.nsc.Global$Run.<init>(Global.scala:1215)
	at scala.tools.nsc.Driver.doCompile(Driver.scala:31)
	at scala.tools.nsc.MainClass.doCompile(Main.scala:23)
	at scala.tools.nsc.Driver.process(Driver.scala:51)
	at scala.tools.nsc.Main.process(Main.scala)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at sbt.compiler.RawCompiler.apply(RawCompiler.scala:26)
	at sbt.compiler.AnalyzingCompiler$$anonfun$compileSources$1$$anonfun$apply$2.apply(AnalyzingCompiler.scala:147)
	at sbt.compiler.AnalyzingCompiler$$anonfun$compileSources$1$$anonfun$apply$2.apply(AnalyzingCompiler.scala:142)
	at sbt.IO$.withTemporaryDirectory(IO.scala:285)
	at sbt.compiler.AnalyzingCompiler$$anonfun$compileSources$1.apply(AnalyzingCompiler.scala:142)
	at sbt.compiler.AnalyzingCompiler$$anonfun$compileSources$1.apply(AnalyzingCompiler.scala:139)
	at sbt.IO$.withTemporaryDirectory(IO.scala:285)
	at sbt.compiler.AnalyzingCompiler$.compileSources(AnalyzingCompiler.scala:139)
	at sbt.compiler.IC$.compileInterfaceJar(IncrementalCompiler.scala:33)
	at com.typesafe.zinc.Compiler$.compilerInterface(Compiler.scala:148)
	at com.typesafe.zinc.Compiler$.create(Compiler.scala:53)
	at com.typesafe.zinc.Compiler.create(Compiler.scala)
	at sbt_inc.SbtIncrementalCompiler.<init>(SbtIncrementalCompiler.java:65)
	at scala_maven.ScalaCompilerSupport.incrementalCompile(ScalaCompilerSupport.java:294)
	at scala_maven.ScalaCompilerSupport.compile(ScalaCompilerSupport.java:119)
	at scala_maven.ScalaCompilerSupport.doExecute(ScalaCompilerSupport.java:99)
	at scala_maven.ScalaMojoSupport.execute(ScalaMojoSupport.java:482)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for zoo-parent 0.10.0-SNAPSHOT:
[INFO] 
[INFO] zoo-parent ......................................... SUCCESS [  0.679 s]
[INFO] analytics-zoo-bigdl_0.12.1-spark_2.4.3 ............. FAILURE [  2.674 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.764 s
[INFO] Finished at: 2021-01-18T00:25:41+07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:3.2.0:compile (scala-compile-first) on project analytics-zoo-bigdl_0.12.1-spark_2.4.3: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:3.2.0:compile failed.: CompileFailed -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :analytics-zoo-bigdl_0.12.1-spark_2.4.3
```

**Comment by beta21s** [comment]
Hi, I really need you to help me.

I really don't know what kind of mistake I'm making.

I have trained the model, but cannot run it with the exported model. While with available models it works.

```
from zoo.common.nncontext import *
from zoo.models.image.objectdetection import *

from zoo.models.image.common.image_config import ImageConfigure
from zoo.feature.image.imagePreprocessing import ImageResize
from zoo.feature.common import ChainedPreprocessing


ImageConfigure(pre_processor=ImageResize(224, 224))
ImageConfigure(pre_processor=ChainedPreprocessing([ImageResize(224, 224), ImageResize(224, 224)]))

sc = init_nncontext()
model = ObjectDetector.load_model("/home/at/Desktop/final.model")
print('done')


# model = ObjectDetector.load_model('final.model')
# image_set = ImageSet.read('img-3.png', sc)
# preprocessing = ChainedPreprocessing(
#                 [ImageResize(256, 256), ImageCenterCrop(224, 224),
#                 ImageChannelNormalize(123.0, 117.0, 104.0), ImageMatToTensor(),
#                 ImageSetToSample()])
# config = ImageConfigure(preprocessing)
# output = model.predict_image_set(image_set, config)


# from zoo.common.nncontext import *
# from zoo.models.image.objectdetection import *

# sc = init_nncontext()
# model = ObjectDetector.load_model(model_path)
# image_set = ImageSet.read(img_path, sc)
# output = model.predict_image_set(image_set)
````

And this is error
```
root@at:/home/at/Desktop# python3 predict.py 
/usr/local/lib/python3.6/dist-packages/bigdl/util/engine.py:41: UserWarning: Find both SPARK_HOME and pyspark. You may need to check whether they match with each other. SPARK_HOME environment variable is set to: /opt/spark-2.4.3, and pyspark is found in: /usr/local/lib/python3.6/dist-packages/pyspark/__init__.py. If they are unmatched, please use one source only to avoid conflict. For example, you can unset SPARK_HOME and use pyspark only.
  warnings.warn(warning_msg)
Prepending /usr/local/lib/python3.6/dist-packages/bigdl/share/conf/spark-bigdl.conf to sys.path
/usr/local/lib/python3.6/dist-packages/zoo/util/engine.py:42: UserWarning: Find both SPARK_HOME and pyspark. You may need to check whether they match with each other. SPARK_HOME environment variable is set to: /opt/spark-2.4.3, and pyspark is found in: /usr/local/lib/python3.6/dist-packages/pyspark/__init__.py. If they are unmatched, you are recommended to use one source only to avoid conflict. For example, you can unset SPARK_HOME and use pyspark only.
  warnings.warn(warning_msg)
Adding /usr/local/lib/python3.6/dist-packages/zoo/share/lib/analytics-zoo-bigdl_0.12.1-spark_2.4.3-0.9.0-jar-with-dependencies.jar to BIGDL_JARS
Prepending /usr/local/lib/python3.6/dist-packages/zoo/share/conf/spark-analytics-zoo.conf to sys.path
creating: createImageResize
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/usr/local/lib/python3.6/dist-packages/zoo/share/lib/analytics-zoo-bigdl_0.12.1-spark_2.4.3-0.9.0-jar-with-dependencies.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/opt/spark-2.4.3/jars/slf4j-log4j12-1.7.16.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
2021-02-05 22:47:58 WARN  Utils:66 - Your hostname, at resolves to a loopback address: 127.0.1.1; using 10.10.100.120 instead (on interface enp0s3)
2021-02-05 22:47:58 WARN  Utils:66 - Set SPARK_LOCAL_IP if you need to bind to another address
2021-02-05 22:47:58 WARN  NativeCodeLoader:62 - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
creating: createImageConfigure
cls.getname: com.intel.analytics.bigdl.python.api.Sample
BigDLBasePickler registering: bigdl.util.common  Sample
cls.getname: com.intel.analytics.bigdl.python.api.EvaluatedResult
BigDLBasePickler registering: bigdl.util.common  EvaluatedResult
cls.getname: com.intel.analytics.bigdl.python.api.JTensor
BigDLBasePickler registering: bigdl.util.common  JTensor
cls.getname: com.intel.analytics.bigdl.python.api.JActivity
BigDLBasePickler registering: bigdl.util.common  JActivity
creating: createImageResize
creating: createImageResize
creating: createChainedPreprocessing
creating: createImageConfigure
creating: createImageResize
creating: createImageCenterCrop
creating: createImageChannelNormalize
creating: createImageMatToTensor
creating: createImageSetToSample
creating: createChainedPreprocessing
creating: createImageConfigure
Traceback (most recent call last):
  File "predict.py", line 20, in <module>
    model = ObjectDetector.load_model("/home/at/Desktop/final.model")
  File "/usr/local/lib/python3.6/dist-packages/zoo/models/image/objectdetection/object_detector.py", line 65, in load_model
    jmodel = callZooFunc(bigdl_type, "loadObjectDetector", path, weight_path)
  File "/usr/local/lib/python3.6/dist-packages/zoo/common/utils.py", line 133, in callZooFunc
    raise e
  File "/usr/local/lib/python3.6/dist-packages/zoo/common/utils.py", line 127, in callZooFunc
    java_result = api(*args)
  File "/usr/local/lib/python3.6/dist-packages/py4j/java_gateway.py", line 1257, in __call__
    answer, self.gateway_client, self.target_id, self.name)
  File "/usr/local/lib/python3.6/dist-packages/py4j/protocol.py", line 328, in get_return_value
    format(target_id, ".", name), value)
py4j.protocol.Py4JJavaError: An error occurred while calling o32.loadObjectDetector.
: java.lang.RuntimeException: Loading module com.intel.analytics.zoo.models.image.objectdetection.ssd.SSDVGG exception :
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.load(ModuleSerializer.scala:154)
	at com.intel.analytics.bigdl.utils.serializer.ModuleLoader$.loadFromFile(ModuleLoader.scala:64)
	at com.intel.analytics.bigdl.nn.Module$.loadModule(Module.scala:61)
	at com.intel.analytics.zoo.models.image.common.ImageModel$.loadModel(ImageModel.scala:132)
	at com.intel.analytics.zoo.models.image.objectdetection.ObjectDetector$.loadModel(ObjectDetector.scala:51)
	at com.intel.analytics.zoo.models.python.PythonZooModel.loadObjectDetector(PythonZooModel.scala:198)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: Loading module com.intel.analytics.bigdl.nn.Sequential exception :
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.load(ModuleSerializer.scala:154)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializable$$anonfun$loadSubModules$1.apply(ModuleSerializable.scala:486)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializable$$anonfun$loadSubModules$1.apply(ModuleSerializable.scala:485)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializable$class.loadSubModules(ModuleSerializable.scala:485)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializer$.loadSubModules(ModuleSerializable.scala:521)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializable$class.doLoadModule(ModuleSerializable.scala:495)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializer$.doLoadModule(ModuleSerializable.scala:521)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializable$class.loadModule(ModuleSerializable.scala:101)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializer$.loadModule(ModuleSerializable.scala:521)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.load(ModuleSerializer.scala:151)
	... 16 more
Caused by: java.lang.RuntimeException: Loading module com.intel.analytics.bigdl.nn.StaticGraph exception :
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.load(ModuleSerializer.scala:154)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializable$$anonfun$loadSubModules$1.apply(ModuleSerializable.scala:486)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializable$$anonfun$loadSubModules$1.apply(ModuleSerializable.scala:485)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializable$class.loadSubModules(ModuleSerializable.scala:485)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializer$.loadSubModules(ModuleSerializable.scala:521)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializable$class.doLoadModule(ModuleSerializable.scala:495)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializer$.doLoadModule(ModuleSerializable.scala:521)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializable$class.loadModule(ModuleSerializable.scala:101)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializer$.loadModule(ModuleSerializable.scala:521)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.load(ModuleSerializer.scala:151)
	... 29 more
Caused by: java.lang.RuntimeException: Loading module com.intel.analytics.bigdl.nn.SpatialConvolution exception :
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.load(ModuleSerializer.scala:154)
	at com.intel.analytics.bigdl.nn.GraphSerializable$$anonfun$prepareLoadModule$3.apply(Graph.scala:586)
	at com.intel.analytics.bigdl.nn.GraphSerializable$$anonfun$prepareLoadModule$3.apply(Graph.scala:585)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at com.intel.analytics.bigdl.nn.GraphSerializable$class.prepareLoadModule(Graph.scala:585)
	at com.intel.analytics.bigdl.nn.Graph$.prepareLoadModule(Graph.scala:476)
	at com.intel.analytics.bigdl.nn.GraphSerializable$class.doLoadModule(Graph.scala:633)
	at com.intel.analytics.bigdl.nn.Graph$.doLoadModule(Graph.scala:476)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializable$class.loadModule(ModuleSerializable.scala:101)
	at com.intel.analytics.bigdl.nn.Graph$.loadModule(Graph.scala:476)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.load(ModuleSerializer.scala:151)
	... 42 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaMethodMirror.jinvokeraw(JavaMirrors.scala:336)
	at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaMethodMirror.jinvoke(JavaMirrors.scala:339)
	at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaVanillaMethodMirror.apply(JavaMirrors.scala:355)
	at com.intel.analytics.bigdl.utils.ReflectionUtils$.reflection(ReflectionUtils.scala:69)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializable$class.doLoadModule(ModuleSerializable.scala:128)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.doLoadModule(ModuleSerializer.scala:36)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializable$class.loadModule(ModuleSerializable.scala:101)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.loadModule(ModuleSerializer.scala:36)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.load(ModuleSerializer.scala:151)
	... 55 more
Caused by: java.lang.OutOfMemoryError: Java heap space
	at scala.reflect.ManifestFactory$$anon$11.newArray(Manifest.scala:133)
	at scala.reflect.ManifestFactory$$anon$11.newArray(Manifest.scala:131)
	at com.intel.analytics.bigdl.tensor.Tensor$.apply(Tensor.scala:963)
	at com.intel.analytics.bigdl.nn.SpatialConvolution.<init>(SpatialConvolution.scala:117)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaMethodMirror.jinvokeraw(JavaMirrors.scala:336)
	at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaMethodMirror.jinvoke(JavaMirrors.scala:339)
	at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaVanillaMethodMirror.apply(JavaMirrors.scala:355)
	at com.intel.analytics.bigdl.utils.ReflectionUtils$.reflection(ReflectionUtils.scala:69)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializable$class.doLoadModule(ModuleSerializable.scala:128)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.doLoadModule(ModuleSerializer.scala:36)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializable$class.loadModule(ModuleSerializable.scala:101)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.loadModule(ModuleSerializer.scala:36)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.load(ModuleSerializer.scala:151)
	at com.intel.analytics.bigdl.nn.GraphSerializable$$anonfun$prepareLoadModule$3.apply(Graph.scala:586)
	at com.intel.analytics.bigdl.nn.GraphSerializable$$anonfun$prepareLoadModule$3.apply(Graph.scala:585)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at com.intel.analytics.bigdl.nn.GraphSerializable$class.prepareLoadModule(Graph.scala:585)
	at com.intel.analytics.bigdl.nn.Graph$.prepareLoadModule(Graph.scala:476)
	at com.intel.analytics.bigdl.nn.GraphSerializable$class.doLoadModule(Graph.scala:633)
	at com.intel.analytics.bigdl.nn.Graph$.doLoadModule(Graph.scala:476)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializable$class.loadModule(ModuleSerializable.scala:101)
	at com.intel.analytics.bigdl.nn.Graph$.loadModule(Graph.scala:476)
	at com.intel.analytics.bigdl.utils.serializer.ModuleSerializer$.load(ModuleSerializer.scala:151)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializable$$anonfun$loadSubModules$1.apply(ModuleSerializable.scala:486)
	at com.intel.analytics.bigdl.utils.serializer.ContainerSerializable$$anonfun$loadSubModules$1.apply(ModuleSerializable.scala:485)

```
